packages/react-scripts/config/webpackDevServer.config.js

```diff
@@ -109,27 +109,31 @@ module.exports = function (proxy, allowedHost) {
     },
     // `proxy` is run between `before` and `after` `webpack-dev-server` hooks
     proxy,
-    onBeforeSetupMiddleware(devServer) {
-      // Keep `evalSourceMapMiddleware`
-      // middlewares before `redirectServedPath` otherwise will not have any effect
-      // This lets us fetch source contents from webpack for the error overlay
-      devServer.app.use(evalSourceMapMiddleware(devServer));
+    setupMiddlewares(middlewares, devServer) {
+      if (!devServer) {
+        throw new Error('webpack-dev-server is not defined');
+      }
 
       if (fs.existsSync(paths.proxySetup)) {
         // This registers user provided middleware for proxy reasons
         require(paths.proxySetup)(devServer.app);
       }
```

Comment:
Does this need updating? Based on my reading of https://create-react-app.dev/docs/proxying-api-requests-in-development/#configuring-the-proxy-manually and https://webpack.js.org/configuration/dev-server/#devserversetupmiddlewares, the intent of `setupMiddlewares` would be that the `middlewares` array is passed to the user's custom proxy logic, instead of passing the `app` for the user's custom logic to call `app.use`.